crates/adapters/dydx/src/python/http.rs

```diff
@@ -0,0 +1,204 @@
+// -------------------------------------------------------------------------------------------------
+//  Copyright (C) 2015-2025 Nautech Systems Pty Ltd. All rights reserved.
+//  https://nautechsystems.io
+//
+//  Licensed under the GNU Lesser General Public License Version 3.0 (the "License");
+//  You may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at https://www.gnu.org/licenses/lgpl-3.0.en.html
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------------------------------
+
+//! Python bindings for dYdX HTTP client.
+
+use std::str::FromStr;
+
+use nautilus_core::python::to_pyvalue_err;
+use nautilus_model::python::instruments::instrument_any_to_pyobject;
+use pyo3::prelude::*;
+use rust_decimal::Decimal;
+use ustr::Ustr;
+
+use crate::http::client::DydxHttpClient;
+
+#[pymethods]
+impl DydxHttpClient {
+    /// Creates a new [`DydxHttpClient`] instance.
+    ///
+    /// # Parameters
+    ///
+    /// - `base_url`: Optional base URL for the HTTP client (defaults to mainnet)
+    /// - `is_testnet`: Whether to use testnet endpoints (default: false)
+    ///
+    /// # Examples
```

Comment:
We don't need examples for internal components like this yet (can be introduced and standardized later on).